### DISCUSSION TITLE
How to access internal setter of LookupClient when doing unit testing?

### DISCUSSION THREAD

**Discussion by henrik73** [sipsorcery-org/sipsorcery#508] [Ideas]
Hi

I have a solution with the SipSourcery  project imported. I need to set the LookupClient in SIPDns and/or the  ResolveSIPUriInternalAsync in SipTransport. The problem is that they are internals, despite there are suggestions they should be set when doing unit testing. 

One solution would be to add [assembly: InternalsVisibleTo("name of assembly here")] in the SipSourcery project, but I really don't want to fork just because of this? Wouldn't it be appropriate to change the setters to be public?

Wouldn't also be more appropriate to use the interface "ILookupClient" to be able to mock?

        public static LookupClient LookupClient
        {
            get
            {
                return _lookupClient;
            }
            internal set
            {
                // Intended to allow unit testing with client options that will cause the
                // lookup logic to execute failure conditions.
                _lookupClient = value;
            }
        }

**Comment by sipsorcery** [comment]
Seem to be good points to me. Will deal with in #509.

**Reply by henrik73** [reply]
Thanks for the fastest ever fix and response!
